lib/python/orchest-internals/_orchest/internals/utils.py

```
@@ -139,3 +139,14 @@ def get_orchest_mounts(project_dir, host_project_dir, mount_form="docker-sdk"):
                 mounts.append(mount)
 
     return mounts
+
+
+def docker_images_list_safe(docker_client, *args, attempt_count=10, **kwargs):
+    while True:
+        attempt_count -= 1
+        try:
```

probably better to just catch the ImageNotFound exception and let it fail otherwise?